src/amplitude-client.js

```diff
@@ -225,6 +227,34 @@ AmplitudeClient.prototype.init = function init(apiKey, opt_userId, opt_config, o
   }
 };
 
+/**
+ * @private
+ */
+AmplitudeClient.prototype._migrateUnsentEvents = function _migrateUnsentEvents(cb) {
+  Promise.all([
+      AsyncStorage.getItem(this.options.unsentKey),
+      AsyncStorage.getItem(this.options.unsentIdentifyKey),
+  ]).then((values) => {
```

Comment:
Are you running this migration every single time the SDK inits? You only need to run it one time, which is when the user loads this new SDK for the first time. Do you need to check if values is null or something for all the subsequent times it's run?? You could also just check that the unsentKey and unsentIdentifyKey exist in Async storage, if not just skip this entire code block?